### PR TITLE
implementation of the remoting error

### PR DESCRIPTION
**Pull request by divath**
Usage example:

```
var ErrorLikeAlreadyGiven = remoting.MustErrorType(
   remoting.ErrorCodeConflict,
   "Facebook:LikeAlreadyGiven",
 )
 ...
 func HandleLikePost(w ResponseWriter, r *Request) {
   ...
   post := getPost(postID)
   if post == nil {
     remoteErr := remoting.NewNotFoundError(
       map[string]string{
         "postId": postID,
       },
     )
     remoting.WriteErrorResponse(w, remoteErr)
     return
   }
   if post.hasLike(userID) {
     remoteErr := remoting.NewError(
       ErrorLikeAlreadyGiven,
       map[string]string{
         "postId": postID,
         "userId": userID,
       },
     )
     remoting.WriteErrorResponse(w, remoteErr)
     return
   }
   ...
 }
```

This change is [Reviewable](https://reviewable.io/reviews/palantir/pkg/99)

### CODE REVIEW

File: remoting/error_instance_id.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+package remoting
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+)
+
+type ErrorInstanceID [16]byte // TODO Can I use existing UUID implementation?
```
Comment (bmoylan) [inline remoting/error_instance_id.go]: This is the most widely used one https://github.com/satori/go.uuid

Comment (divath) [reply remoting/error_instance_id.go]: @nmiyake and suggestions? I think there is also google uuid package that is popular.

Comment (bmoylan) [reply remoting/error_instance_id.go]: Kubernetes uses this one https://github.com/pborman/uuid
This is the google one https://github.com/google/uuid

Comment (divath) [reply remoting/error_instance_id.go]: I guess it does not matter that much as long as I am not exposing these types directly.

Comment (divath) [reply remoting/error_instance_id.go]: using `google/uuid` for now, it is not exported via the `remoting` package, so it should not be a big deal

**Comment by divath** [comment]
@bmoylan @nmiyake any updates?

**Comment by divath** [comment]
Moving to other repository.

**Pull request closed without merging**